pkg/k8s/K8sCommonService.go

```diff
@@ -342,3 +343,112 @@ func (impl *K8sCommonServiceImpl) GetCoreClientByClusterId(clusterId int) (*kube
 	}
 	return clientSet, v1Client, nil
 }
+
+func (impl K8sCommonServiceImpl) PortNumberExtraction(resp []BatchResourceResponse, resourceTree map[string]interface{}, err error) (map[string]interface{}, error) {
+	portsService := make([]int64, 0)
+	portsEndpoint := make([]int64, 0)
+	portEndpointSlice := make([]int64, 0)
+	for _, portHolder := range resp {
+		if portHolder.ManifestResponse == nil {
+			continue
+		}
+		kind, ok := portHolder.ManifestResponse.Manifest.Object["kind"]
+		if !ok {
+			impl.logger.Errorw("kind not found in resource tree, unable to extract port no")
+			return resourceTree, nil
+		}
+		if kind == ServiceKind {
+			specField, ok := portHolder.ManifestResponse.Manifest.Object["spec"]
+			if !ok {
+				impl.logger.Errorw("spec not found in resource tree, unable to extract port no")
+				return resourceTree, nil
+			}
+			spec := specField.(map[string]interface{})
+			if spec != nil {
+				ports, ok := spec["ports"]
+				if !ok {
+					impl.logger.Errorw("ports not found in resource tree, unable to extract port no")
+					return resourceTree, nil
+				}
+				portList := ports.([]interface{})
+				for _, portItem := range portList {
+					if portItem.(map[string]interface{}) != nil {
+						portNumbers := portItem.(map[string]interface{})["port"]
```

Comment:
let's add ok bool validation for every type assertion. example - 
portNumbers := portItem.(map[string]interface{})["port"] --> portNumbers, ok := portItem.(map[string]interface{})["port"]